### PR TITLE
Implements min and max functions in the ast

### LINKED ISSUE #24: [LCGS] Add parsing of min and max operators
The user should be able to write the expressions `max(e1, e2)` and `min(e1, e2)` where `e1` and `e2` are expressions.

We don't have functions so the `max` and `min` should be considered operators that takes *n* arguments. The `min` and `max` operators are primary expressions and should therefore be parsed in `src/lcgs/parse.rs::primary_expr()` and a `Min` and `Max` variant needs to be added to `src/lcgs/ast.rs::ExprKind`.

### PR DESCRIPTION
**Pull request by GeneralDenmark**
Fixes #24

### CODE REVIEW

**Comment by GeneralDenmark** [comment]
fixes #24

File: src/lcgs/ir/eval.rs [hunk 1]
```diff
@@ -21,6 +21,8 @@ impl<'a> Evaluator<'a> {
             ExprKind::UnaryOp(op, e) => self.eval_unop(op, e),
             ExprKind::BinaryOp(op, e1, e2) => self.eval_binop(op, e1, e2),
             ExprKind::TernaryIf(c, e1, e2) => self.eval_if(c, e1, e2),
+            ExprKind::Max(op) => self.eval_min(op),
+            ExprKind::Min(op) => self.eval_max(op),
```
Comment (NicEastvillage) [inline src/lcgs/ir/eval.rs:25]: `op`? more like `exprs`

File: src/lcgs/ir/symbol_checker.rs [hunk 2]
```diff
@@ -186,4 +191,82 @@ impl<'a> SymbolChecker<'a> {
...
+                    kind: ExprKind::Min(res),
+                })
+            }
+        } else {
+            Ok(Expr {
+                kind: ExprKind::Min(
+                    checked_list
+                        .iter()
+                        .map(|p| self.check(p).unwrap())
+                        .collect(),
```
Comment (NicEastvillage) [inline src/lcgs/ir/symbol_checker.rs:228]: The expressions in `checked_list` are already checked. Same problem in `check_max`.

File: src/lcgs/ir/eval.rs [hunk 3]
```diff
@@ -64,4 +66,10 @@ impl<'a> Evaluator<'a> {
             self.eval(e2)
         }
     }
+    fn eval_min(&self, ls: &Vec<Expr>) -> Result<i32, ()> {
+        return ls.iter().map(|p| self.eval(p)).min().unwrap();
+    }
+    fn eval_max(&self, ls: &Vec<Expr>) -> Result<i32, ()> {
+        return ls.iter().map(|p| self.eval(p)).max().unwrap();
+    }
```
Comment (NicEastvillage) [inline src/lcgs/ir/eval.rs:74]: Looks right to me.

**Review by NicEastvillage — CHANGES_REQUESTED** [review]

File: src/lcgs/ir/eval.rs (hunk 1 above)
Comment (cogitantium) [reply src/lcgs/ir/eval.rs]: 5f29b4d

File: src/lcgs/ir/symbol_checker.rs (hunk 2 above)
Comment (cogitantium) [reply src/lcgs/ir/symbol_checker.rs]: 5f29b4d

**Review by cogitantium — APPROVED** [review]
LGTM

**Review by NicEastvillage — APPROVED** [review]
Noice

**Review by jenrik — CHANGES_REQUESTED** [review]
Some `clippy` warnings needs to be fixed

**Review by jenrik — APPROVED** [review]

**Review by NicEastvillage — APPROVED** [review]

**Pull request merged**